src/algebra/lie/basic.lean

```diff
@@ -176,6 +176,32 @@ instance : lie_module R L (M →ₗ[R] N) :=
 
 end basic_properties
 
+section non_unital_non_assoc_semiring
+
+variables (R : Type u) (L : Type v) [comm_ring R] [lie_ring L] [lie_algebra R L]
+
+/-- A `lie_ring` can be regarded as a `non_unital_non_assoc_semiring` by turning its
+`has_bracket` (denoted `⁅, ⁆`) into a `has_mul` (denoted `*`). -/
+def lie_ring.to_non_unital_non_assoc_semiring : non_unital_non_assoc_semiring L :=
+{ mul           := has_bracket.bracket,
+  left_distrib  := lie_add,
+  right_distrib := add_lie,
+  zero_mul      := zero_lie,
+  mul_zero      := lie_zero,
+  .. (infer_instance : add_comm_monoid L) }
+
+local attribute [instance] lie_ring.to_non_unital_non_assoc_semiring
+
+/-- Regarding the `lie_ring` of a `lie_algebra` as a `non_unital_non_assoc_semiring`, we can
+reinterpret the `smul_lie` law as an `is_scalar_tower`. -/
+lemma lie_algebra.is_scalar_tower : is_scalar_tower R L L := ⟨smul_lie⟩
```

Comment:
I think this is safe to make an instance

Reply:
To expand on this; these are instances about the definition `lie_ring.to_non_unital_non_assoc_semiring`, and they live in Prop, so there's no danger of them applying when you don't want them to (as there is no situation where an additional lemma being true is bad)

Reply:
I wondered this too.

Perhaps it's fine but what worries me is the `lie_algebra` associated to an `algebra` via [this instance](https://github.com/leanprover-community/mathlib/blob/2df573a507e1f458f03bbbcf3330b071de2d3478/src/algebra/lie/of_associative.lean#L76) (or indeed, more generally if we one day define Poisson algebras). We have two different instances of `is_scalar_tower R A A`, namely:

- `lie_algebra.is_scalar_tower R A`
- `@is_scalar_tower.right R A _ _ _`

(and similarly for `smul_comm_class`)

Reply:
I think this is OK after all.